src/iuse.cpp

```diff
@@ -5376,6 +5376,11 @@ int iuse::hotplate( player *p, item *it, bool, const tripoint & )
 }
 
 int iuse::towel( player *p, item *it, bool t, const tripoint & )
+{
+    return towel_common( p, it, t );
+}
+
+int iuse::towel_common( player *p, item *it, bool t )
 {
     if( t ) {
```

Comment:
This block should only be in ` iuse::towel` (it makes no sense when called from the vehicle code, and the `t` parameter from there is not needed either).

Similar the function should return `void`, Determining how many charges are consumed can be left to the code that actually has an item (`iuse::towel`), not necessarily the vehicle code.

Reply:
The vehicle code should have charges too (now it soaks unlimited amount of shit for free), but i left it to be implemented someday™.